e2e/preferences.e2e.spec.ts

```diff
@@ -0,0 +1,144 @@
+import os from 'os';
+import path from 'path';
+import { ElectronApplication, BrowserContext, _electron, Page } from 'playwright';
+import { test, expect } from '@playwright/test';
+import { createDefaultSettings, packageLogs, reportAsset } from './utils/TestUtils';
+import { NavPage } from './pages/nav-page';
+import { PreferencesPage } from './pages/preferences';
+
+let page: Page;
+
+/**
+ * Using test.describe.serial make the test execute step by step, as described on each `test()` order
+ * Playwright executes test in parallel by default and it will not work for our app backend loading process.
+ * */
+test.describe.serial('Main App Test', () => {
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+  let preferencesWindow: Page;
+
+  test.beforeAll(async() => {
+    createDefaultSettings();
+
+    electronApp = await _electron.launch({
+      args: [
+        path.join(__dirname, '../'),
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        // See src/utils/commandLine.ts before changing the next item as the final option.
+        '--disable-dev-shm-usage',
+        '--no-modal-dialogs',
+      ],
+      env: {
+        ...process.env,
+        RD_LOGS_DIR: reportAsset(__filename, 'log'),
+      },
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+
+    new NavPage(page).preferencesButton.click();
+
+    await electronApp.waitForEvent('window');
+    await electronApp.waitForEvent('window');
```

Comment:
We need the two events because there's the third "window" for `chrome-extension://nphilambcliphnhecoecnbigaejdnopp/_generated_background_page.html` that is raised before preferences. Only waiting for the window event once does not give us an accurate representation of what we need. I can make a comment on this for future contributors.

Reply:
Right, which is why this `waitForEvent` has a filter to test the page URL.
(I think I ended up testing it later, but mixed up which browser window had the current review.)

Reply:
Fair enough. We do end up awaiting both window events with this approach, but debugging reveals that this can happen out of sequence (i.e. we get our preferences window after the first test runs).

I'd like to reintroduce the assertion the assertion `    expect(preferencesWindow?.url().toLowerCase()).toContain('preferences');` to guard against confusion in the event of a failure.